zp-relayer/workers/sentTxWorker.ts

```diff
@@ -124,81 +118,95 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
+        const rescheduledIds = await Promise.all(reschedulePromises)
+
+        return [SentTxState.REVERT, txHash, rescheduledIds] as SentTxResult
       }
     } else {
       // Resend with updated gas price
-      const txConfig = job.data.txConfig
+      const fetchedGasPrice = await gasPrice.fetchOnce()
+      const oldWithExtra = addExtraGasPrice(lastGasPrice, config.minGasPriceBumpFactor, null)
+      const newWithExtra = addExtraGasPrice(fetchedGasPrice, config.gasPriceSurplus, null)
 
-      const oldGasPrice = job.data.gasPriceOptions
-      const fetchedGasPrice = gasPrice.getPrice()
+      const newGasPrice = chooseGasPriceOptions(oldWithExtra, newWithExtra)
 
-      const newGasPrice = chooseGasPriceOptions(oldGasPrice, fetchedGasPrice)
-
-      logger.warn('Tx %s is not mined; updating gasPrice: %o -> %o', txHash, oldGasPrice, newGasPrice)
+      logger.warn('%s Tx %s is not mined; updating gasPrice: %o -> %o', logPrefix, lastHash, lastGasPrice, newGasPrice)
 
       const newTxConfig = {
         ...txConfig,
         ...newGasPrice,
       }
 
+      let newTxHash
       try {
-        if (txType === TxType.PERMITTABLE_DEPOSIT) {
-          const deadline = (txData as PermittableDepositTxData).deadline
-          await checkAssertion(() => checkDeadline(toBN(deadline), config.permitDeadlineThresholdResend))
-        }
-
-        const newTxHash = await signAndSend(newTxConfig, config.relayerPrivateKey, web3)
-
-        // Add updated job
-        await sentTxQueue.add(
-          newTxHash,
-          {
-            ...job.data,
-            txHash: newTxHash,
-            txConfig: newTxConfig,
-            gasPriceOptions: newGasPrice,
-          },
-          {
-            priority: txConfig.nonce,
-            delay: config.sentTxDelay,
-          }
-        )
-        return [SentTxState.RESEND, newTxHash] as SentTxResult
+        newTxHash = await signAndSend(newTxConfig, config.relayerPrivateKey, web3)
       } catch (e) {
         const err = e as Error
-        logger.warn('%s: Tx resend failed for %s: %s', logPrefix, txHash, err.message)
-        if (isSameTransactionError(err) || isGasPriceError(err)) {
-          // Force update gas price
-          gasPrice.start()
-        } else if (isNonceError(err)) {
-          await updateNonce(await readNonce(true))
-        } else {
-          // Error can't be handled
-          logger.error('%s: Error cannot be handled: %o', logPrefix, err)
-          // Rollback the tree
-          await clearOptimisticState(redis)
-          return [SentTxState.FAILED, txHash] as SentTxResult
+        logger.warn('%s Tx resend failed for %s: %s', logPrefix, lastHash, err.message)
+        if (isGasPriceError(err) || isSameTransactionError(err)) {
+          // Tx wasn't sent successfully, but still update last attempt's
+          // gasPrice to be acccounted in the next iteration
+          job.data.prevAttempts[job.data.prevAttempts.length - 1][1] = newGasPrice
```

Comment:
Also thought about it. But it will increase redundant transaction requests to RPC. If we want to add a new record, maybe we should filter repeating hashes when calling a node?

Reply:
New hash is returned by `signAndSend` only if tx submission was successful. If promise was rejected we will get an error. Do you think we should still save new hash in this case, as tx wasn't accepted by a node?

Reply:
I think it is ok to save this hash.
In case of `isSameTransactionError`, the tx with such hash is already in the mempool, so it is even better if we save it.
And `isGasPriceError` should not really happen, if we configure everything properly.